app/controllers/api/v0/sources_controller.rb

```diff
@@ -6,6 +6,17 @@ class SourcesController < ApplicationController
       include Api::V0::Mixins::ShowMixin
       include Api::V0::Mixins::UpdateMixin
 
+      def with_current_request
+        ManageIQ::API::Common::Request.with_request(request) do |current|
+          if Tenant.tenancy_enabled?
+            # Auto-create tenant if creating a new source and tenant does not exist
+            Tenant.find_or_create_by(:external_tenant => current.user.tenant) if action_name == "create" rescue KeyError
```

Comment:
Do you need to set the current tenant here after creating it?

Reply:
Should be handled by `super`. But, in looking at that, I saw the the test was not quite correct.